src/index.test.ts

```diff
@@ -317,6 +317,51 @@ test('Should bootstrap data when bootstrap is provided', async () => {
     expect(localStorage.getItem(storeKey)).toBe(JSON.stringify(bootstrap));
 });
 
+it('should return correct variant if called asynchronously multiple times', async () => {
+    const bootstrap = [
+        {
+            name: 'foo',
+            enabled: true,
+            variant: {
+                name: 'A',
+                enabled: true,
+                payload: {
+                    type: 'string',
+                    value: 'FOO',
+                },
+            },
+            impressionData: false,
+        },
+    ];
+
+    const config: IConfig = {
+        url: 'http://localhost/test',
+        clientKey: '12',
+        appName: 'web',
+        refreshInterval: 0,
+        metricsInterval: 0,
+        disableRefresh: true,
+        bootstrapOverride: true,
+        bootstrap,
+        createAbortController: () => null,
+    };
+    const client = new UnleashClient(config);
+
+    for (let i = 0; i < 12; i++) {
+        await true;
```

Comment:
what's this do? awaiting a sync variable seems pointless, but I'm guessing you put it here for some reason?

Reply:
it's repro of this issue: https://github.com/Unleash/unleash-client-nextjs/issues/106

Reply:
Yes. This is important for the bug to occur - it will effectively wrap everything in a promise and add it to a [microtask queue](https://javascript.info/microtask-queue).